### PR TITLE
Add recipe for gz-cmake

### PR DESCRIPTION
**Pull request by traversaro**
The ignition robotics libraries, that are currently packaged in conda-forge under the `libignition-*`, have been renamed to Gazebo libraries (see https://community.gazebosim.org/t/a-new-era-for-gazebo/1356).

To avoid confusion related to the feedstock name in the future, we are re-submitting all the `libignition-` recipes to conda-forge to have descriptive and coherent names that start with `gz-`, as described in https://github.com/conda-forge/libignition-cmake0-feedstock/issues/26#issuecomment-1255871405 . 

This the first PR, that create a new `gz-cmake-feedstock` to package [`gz-cmake`](https://github.com/gazebosim/gz-cmake), using the recipe already available in https://github.com/conda-forge/libignition-cmake0-feedstock .

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
@wolfv @j-rivero are you interested in remaining mantainers of the feedstock?

**Comment by traversaro** [comment]
> @wolfv @j-rivero are you interested in remaining mantainers of the feedstock?

Friendly ping @wolfv @j-rivero on this, so we can proceed with the other recipes as discussed on https://github.com/conda-forge/libignition-cmake0-feedstock/issues/26 .

**Comment by traversaro** [comment]
> > @wolfv @j-rivero are you interested in remaining mantainers of the feedstock?
> 
> Friendly ping @wolfv @j-rivero on this, so we can proceed with the other recipes as discussed on [conda-forge/libignition-cmake0-feedstock#26](https://github.com/conda-forge/libignition-cmake0-feedstock/issues/26) , thanks!

I removed you to speed up the review process, feel free to add yourself back once the PR is merged or even while waiting for a review, thanks!

**Comment by traversaro** [comment]
@conda-forge/help-c-cpp The recipe is ready for review.

**Comment by j-rivero** [comment]
ouch sorry for the silence Silvio, just saw the messages. It is fine to me to go with whatever is faster at this point. Reading all the changes they look good as fast as my understanding of conda arrives.

**Review by j-rivero — APPROVED** [review]

**Comment by wolfv** [comment]
Sorry Silvio, I'll give you blanket permission to add me to any feedstocks related to this :)

**Comment by traversaro** [comment]
> Sorry Silvio, I'll give you blanket permission to add me to any feedstocks related to this :)

Great!

**Comment by traversaro** [comment]
> ouch sorry for the silence Silvio, just saw the messages. It is fine to me to go with whatever is faster at this point. Reading all the changes they look good as fast as my understanding of conda arrives.

No problem, I will add you back once the feedstock is created!

**Pull request merged**
